main.go

```diff
@@ -61,12 +61,25 @@ func setupCollectors(logger log.Logger, configFile string) ([]prometheus.Collect
 	level.Info(logger).Log("msg", "Configuring route53 with region", "region", config.Route53Config.Region)
 	level.Info(logger).Log("msg", "Will VPC metrics be gathered?", "vpc-enabled", config.VpcConfig.Enabled)
 	// Create a single session here, because we need the accountid, before we create the other configs
+	roleARN := os.Getenv("ROLE_ARN")
+	sessionName := os.Getenv("SESSION_NAME")
 	awsConfig := aws.NewConfig().WithRegion("us-east-1")
 	sess := session.Must(session.NewSession(awsConfig))
 	awsAccountId, err := getAwsAccountNumber(logger, sess)
 	if err != nil {
 		return collectors, err
 	}
+	if pkg.LookUpEnvVar("ROLE_ARN") && pkg.LookUpEnvVar("SESSION_NAME") {
+		client := sts.New(sess)
+		err = pkg.AssumeRole(client, roleARN, sessionName, logger)
```

Comment:
In order to refresh the token/session, first you need to introduce the [DurationSeconds](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) Parameters for `AssumeRole`. So we can know how often we should refresh the token. Then, we need to write a loop function that looks like [CollectLoop](https://github.com/app-sre/aws-resource-exporter/blob/master/pkg/vpc.go#L121-L137) which would run `AssumeRole` before DurationSeconds comes and recreate all sessions in `vpcSessions`, `rdsSessions`, `ec2Sessions` and r53 session.